packages/proxy/lib/http/request-middleware.ts

```diff
@@ -21,13 +22,36 @@ const LogRequest: RequestMiddleware = function () {
   this.next()
 }
 
-const ExtractIsAUTFrameHeader: RequestMiddleware = function () {
+const ExtractCypressMetadataHeaders: RequestMiddleware = function () {
   this.req.isAUTFrame = !!this.req.headers['x-cypress-is-aut-frame']
+  const requestIsXhrOrFetch = this.req.headers['x-cypress-request']
```

Comment:
I feel like `x-cypress-request` is a little too generic a name for this header. What about `x-cypress-is-xhr-or-fetch`?

Reply:
I think that's a good idea. Will update!

Reply:
I'm leaning towards `x-cypress-is-xhr-or-fetch` since in electron/firefox it could be `true` instead of just `xhr` or `fetch`, which I think encapsulates all 3 possibilities.

Reply:
addressed in #24060